spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/test/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/KubernetesModuleDeployerIT.java

```diff
@@ -0,0 +1,120 @@
+package org.springframework.cloud.dataflow.module.deployer.kubernetes;
+
+import static org.junit.Assert.*;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.dataflow.core.ModuleCoordinates;
+import org.springframework.cloud.dataflow.core.ModuleDefinition;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentId;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentRequest;
+import org.springframework.cloud.dataflow.module.ModuleInstanceStatus;
+import org.springframework.cloud.dataflow.module.ModuleStatus;
+import org.springframework.cloud.dataflow.module.ModuleStatus.State;
+import org.springframework.cloud.dataflow.module.deployer.ModuleDeployer;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = KubernetesModuleDeployerConfiguration.class)
+public class KubernetesModuleDeployerIT {
+	
+	@Autowired
+	private ModuleDeployer processModuleDeployer;
+	
+	@Autowired
+	private KubernetesClient kubeClient;
+	
+	private final int TIMEOUT = 30 * 1000; // 30 seconds
+	
+	private ExecutorService executor; 
+
+	@Before 
+	public void setup() {
+		assertNotNull(processModuleDeployer);
+		assertNotNull(kubeClient);
+		executor = Executors.newSingleThreadExecutor();
+	}
+	
+	@After
+	public void teardown() {
+		executor.shutdownNow();
+	}
+	
+	@Test
+	public void end2endDeployment1() throws InterruptedException, ExecutionException  {
+		
+		String group = "deployment-test-0";
+		String label = "http";
+		
+		ModuleDefinition d = new ModuleDefinition.Builder()
+				.setName("foobar")
```

Comment:
label and name are actually semantically swapped in your test. Does not matter given that you force the maven coordinates anyway, but what you meant is most certainly the other way around (or there is a bug in s-c-d)